src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -99,19 +99,19 @@ class primitive_inst {
     }
     memory& output_memory(size_t index = 0) const { return *_outputs[index]; }
     memory::ptr output_memory_ptr(size_t index = 0) const { return _outputs[index]; }
-    size_t inputs_memory_count() const { return _node->get_primitive()->input_size(); }
-    size_t outputs_memory_count() const { return _node->get_primitive()->output_size(); }
+    size_t inputs_memory_count() const { return _inputs_memory_count; }
+    size_t outputs_memory_count() const { return _outputs_memory_count; }
     bool outputs_allocated() const {
         if (_outputs.empty()) return false;
         for (const auto& output : _outputs) {
             if (!output) return false;
         }
         return true;
     }
-    primitive_type_id type() const { return _node->type(); }
-    primitive_id id() const { return _node->id(); }
-    primitive_id org_id() const { return _node->get_org_primitive_id(); }
-    bool can_be_optimized() const { return _node->can_be_optimized(); }
+    primitive_type_id type() const { return _type; }
+    primitive_id id() const { return _id; }
+    primitive_id org_id() const { return _org_id; }
+    bool can_be_optimized() const { return _can_be_optimized; }
```

Comment:
Hmm BTW for deserialized network from a blob, do we need to restore "optimized" inst?
Maybe we dont need to serialize/deserialize those inst.

Reply:
Because the `can_be_optimized()` check function is called in all insts, it is needed.

Reply:
I am not about the variable. I meant the need of optimized inst, in the  serialize/deserialize :) 
(which is not to be handled in this PR, but in your upcoming work)

Reply:
Though optimized insts are not executed, they are still necessary when building dependency between insts in the current design.

Reply:
hmm if it is only for the dep, sounds like that we can get rid of them in the future. Anyway it is not for now.